### PR TITLE
Feature/popups for creating user and connection

### PR DESCRIPTION
**Pull request by delieva**
Everything is done excepting rerender count of users/connections and table/grid after successful creation

### CODE REVIEW

File: src/components/User/newUserPopup/newUserPopup.module.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+.select {
+  width: 100%;
+  height: 30px;
+  background-color: white;
+  -webkit-appearance: none;
+  -moz-appearance: none;
+  padding-left: 5px;
+  font: normal 18px "Montserrat";
```
Comment (VladaZhmurkevych) [inline src/components/User/newUserPopup/newUserPopup.module.scss]: We have variables for fonts, so it is no needed to write this by yourself. Is there is no font-size which is needed for you, it is better to use a variable, which sets you font-bold and font-family, and write here font-size by yourself

File: src/components/User/newUserPopup/newUserPopup.module.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  visibility: hidden;
+  height: 2em;
+}
+
+.select {
+  width: 100%;
+  height: 30px;
+  background-color: white;
+  -webkit-appearance: none;
+  -moz-appearance: none;
```
Comment (VladaZhmurkevych) [inline src/components/User/newUserPopup/newUserPopup.module.scss]: We have autoprefixer, so you do need to write this

File: src/components/shared/BasePopup/index.js [hunk 3]
```diff
@@ -33,9 +34,9 @@ const BasePopup = ({
       classNames="alert"
       unmountOnExit
     >
-      <div className={styles.popupWrapper}>
+      <div className={classnames(styles.popupWrapper, popupClassName)}>
```
Comment (VladaZhmurkevych) [inline src/components/shared/BasePopup/index.js:37]: I can't find popupClassName. Where is it declared?

**Review by VladaZhmurkevych — CHANGES_REQUESTED** [review]

File: src/store/createUser/reducer.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    isLoading: true
+  },
+  currentPage: 1,
+  search: '',
+  itemsPerPage: 4,
+  lastLoadedPage: 1,
+  error: false,
+  isLoading: false,
+  hasNextPage: true,
+  users: [],
```
Comment (MishaHnatyshyn) [inline src/store/createUser/reducer.js]: Remove these fields, expects error from createUser reducer as it does not relate to creating user logic

File: src/store/createUser/reducer.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        errorMessage: 'Creation failed'
+      };
+    case CREATE_SUCCESS:
+      return {
+        ...state,
+        users: [action.payload, state.users],
+        totalUsers: {
+          count: state.totalUsers.count + 1,
+          isLoading: false
+        },
```
Comment (MishaHnatyshyn) [inline src/store/createUser/reducer.js]: move this logic to adminUsers reducer and leave here only cleaning of form

File: src/store/connectionForm/reducer.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      return {
+        ...state,
+        error: true,
+        errorMessage: EMPTY_FIELDS_ERROR_MESSAGE
+      };
+    case APPEND_CONNECTIONS:
+      return {
+        ...state,
+        connections: [...state.connections, action.payload],
+        hasNextPage: action.payload.length > 0,
```
Comment (MishaHnatyshyn) [inline src/store/connectionForm/reducer.js]: move this logic to connections reducer and leave here only form cleaning

File: src/store/connectionForm/actions.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        host,
+        port,
+        name: nameConnection,
+        databaseName: nameDB,
+        username: name,
+        password,
+        typeId: Number(type),
+      }
+    });
+    dispatch(onCloseAction());
```
Comment (MishaHnatyshyn) [inline src/store/connectionForm/actions.js]: What does it do ?

File: src/store/connection/reducer.js [hunk 8]
```diff
@@ -76,7 +76,7 @@ export default function connectionsReducer(state = initialState, action) {
     case APPEND_CONNECTIONS:
       return {
         ...state,
-        connections: [...state.connections, ...action.payload],
+        connections: [...state.connections, action.payload],
```
Comment (MishaHnatyshyn) [inline src/store/connection/reducer.js]: I believe we should add new conenction to the start of list: `connections: [ action.payload, ...state.connections],`

File: src/store/connection/actions.js [hunk 9]
```diff
@@ -54,6 +55,7 @@ export const searchConnections = () => async (dispatch, getState) => {
   try {
     const data = await get(ADMIN_CONNECTIONS_ENDPOINT, { params });
     dispatch(setConnections(data));
+    dispatch(push('/admin/databases'));
```
Comment (MishaHnatyshyn) [inline src/store/connection/actions.js]: Why do we need this ?

File: src/store/adminUsers/reducer.js [hunk 10]
```diff
@@ -137,8 +137,12 @@ export default function adminUsersReducer(state = initialState, action) {
...
-        users: [action.payload, ...state.users],
+        users: [action.payload, state.users],
         isCreatingInProgress: false,
+        totalUsers: {
+          count: state.totalUsers.count + 1,
+          isLoading: false
+        },
         formUsername: '',
         formPassword: '',
         formPasswordRepeat: '',
```
Comment (MishaHnatyshyn) [inline src/store/adminUsers/reducer.js]: remove this logic from here

File: src/store/adminUsers/reducer.js [hunk 11]
```diff
@@ -137,8 +137,12 @@ export default function adminUsersReducer(state = initialState, action) {
     case CREATE_SUCCESS:
       return {
         ...state,
-        users: [action.payload, ...state.users],
+        users: [action.payload, state.users],
```
Comment (MishaHnatyshyn) [inline src/store/adminUsers/reducer.js]: `users: [action.payload, ...state.users],`

File: src/store/adminUsers/reducer.js [hunk 12]
```diff
@@ -29,14 +29,13 @@ const initialState = {
...
-    formDescription: '',
-    isCreatingInProgress: false,
-  },
+  formUsername: '',
+  formPassword: '',
+  formPasswordRepeat: '',
+  formUserType: '',
+  formDescription: '',
+  isCreatingInProgress: false,
+  isVisible: true,
```
Comment (MishaHnatyshyn) [inline src/store/adminUsers/reducer.js:33]: remove this from here

File: src/pages/page.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+import PropTypes from 'prop-types';
+import NewUserPopup from '../components/User/newUserPopup';
+import NewConnection from '../components/newDBConnectionPopup';
+const UsersAdmin = () => (
+  <NewUserPopup />
+);
+
+
+
+export default UsersAdmin;
```
Comment (MishaHnatyshyn) [inline src/pages/page.js]: what is this ?

File: src/components/User/newUserPopup/newUserPopup.module.scss [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+  padding-left: 5px;
+  font: normal 18px "Montserrat";
+  margin: 20px 0;
+}
+
+.maxWidth {
+  width: 90%;
+}
```
Comment (MishaHnatyshyn) [inline src/components/User/newUserPopup/newUserPopup.module.scss]: This file is almost the same as newDBConnectionPopup.module.scss.

File: src/components/shared/BasePopup/index.js (hunk 3 above)
Comment (delieva) [reply src/components/shared/BasePopup/index.js:37]: It`s a prop

Comment (VladaZhmurkevych) [reply src/components/shared/BasePopup/index.js:37]: okay, thanks :)

File: src/components/User/newUserPopup/NewUserForm/index.js [hunk 15]
```diff
@@ -0,0 +1,125 @@
+import React, { useMemo } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import styles from '../newUserPopup.module.scss';
+import Input from '../../../shared/Input';
+import AlertMessage from '../../../shared/AlertMessage';
+import Textarea from '../../../shared/Textarea';
+import '../../../shared/BasePopup/styles.scss';
```
Comment (VladaZhmurkevych) [inline src/components/User/newUserPopup/NewUserForm/index.js]: You shouldn't import styles from other components.

File: src/components/User/newUserPopup/NewUserForm/index.js [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+            <img src="/images/report.png" alt="error message" />
+          </AlertMessage>
+        </label>
+        <Input id="formPasswordRepeat" type="password" name="formPasswordRepeat" onChange={changePasswordRepeat} value={formPasswordRepeat} />
+        <label className={styles.label} htmlFor="formUserType">USER TYPE</label>
+        <select id="formUserType" className={styles.select} value={formUserType} onChange={changeUserType}>
+          <option value="">user type</option>
+          <option value="1">admin</option>
+          <option value="2">user</option>
+        </select>
```
Comment (VladaZhmurkevych) [inline src/components/User/newUserPopup/NewUserForm/index.js]: use, please, Select component from 'shared'

File: src/components/User/newUserPopup/index.js [hunk 17]
```diff
@@ -0,0 +1,55 @@
+import React, { useMemo } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { preventDefaultHandler } from '../../../utils';
+import '../../shared/BasePopup/styles.scss';
```
Comment (VladaZhmurkevych) [inline src/components/User/newUserPopup/index.js]: the same, don't import these styles

File: src/components/newDBConnectionPopup/NewDBConnectionForm/index.js [hunk 18]
```diff
@@ -0,0 +1,126 @@
+import React, { useMemo } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Input from '../../shared/Input';
+import AlertMessage from '../../shared/AlertMessage';
+import styles from '../../User/newUserPopup/newUserPopup.module.scss';
+import '../../shared/BasePopup/styles.scss';
```
Comment (VladaZhmurkevych) [inline src/components/newDBConnectionPopup/NewDBConnectionForm/index.js]: same here

File: src/components/shared/BasePopup/components/PopupButton/popupButton.module.scss [hunk 19]
```diff
@@ -2,7 +2,7 @@
 
 .button {
   padding: 10px 0;
-  margin: 0 15px;
+  margin: 0 44px;
```
Comment (VladaZhmurkevych) [inline src/components/shared/BasePopup/components/PopupButton/popupButton.module.scss]: Won't it affect other popups?

**Review by VladaZhmurkevych — CHANGES_REQUESTED** [review]

File: src/components/Login/LoginForm/index.js [hunk 20]
```diff
@@ -32,10 +32,10 @@ const LoginForm = ({
       <div className={styles.line} />
       <form onSubmit={formHandler}>
         <div className={styles.inputFields}>
-          <Input text="Username" type="text" onChange={changeUsername} value={username}>
+          <Input text="Username" type="text" withImage onChange={changeUsername} value={username}>
             <img src="/images/user.png" alt="user icon" />
           </Input>
-          <Input text="Password" type="password" onChange={changePassword} value={password}>
+          <Input text="Password" type="password" withImage classes="withImage" onChange={changePassword} value={password}>
```
Comment (VladaZhmurkevych) [inline src/components/Login/LoginForm/index.js]: why do you need `classes withImage`?

File: src/components/User/newUserPopup/NewUserForm/index.js [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+        <Input id="formPassword" type="password" name="password" onChange={changePassword} value={formPassword} />
+        <label className={styles.label} htmlFor="formPasswordRepeat">
+          REPEAT PASSWORD
+          <AlertMessage classes={alertClassesPassRepeat}>
+            <img src="/images/report.png" alt="error message" />
+          </AlertMessage>
+        </label>
+        <Input id="formPasswordRepeat" type="password" name="formPasswordRepeat" onChange={changePasswordRepeat} value={formPasswordRepeat} />
+        <label className={styles.label} htmlFor="formUserType">USER TYPE</label>
+        <Select options={[{ name: 'admin', value: '1' }, { name: 'user', value: '2' }]} classes={styles.select} value={formUserType} onChange={changeUserType} />
```
Comment (VladaZhmurkevych) [inline src/components/User/newUserPopup/NewUserForm/index.js]: move options value to variable

**Review by VladaZhmurkevych — CHANGES_REQUESTED** [review]

File: src/components/newDBConnectionPopup/NewDBConnectionForm/index.js [hunk 22]
```diff
@@ -0,0 +1,124 @@
...
+          <label className={styles.label} htmlFor="nameDBField">DATABASE NAME</label>
+          <Input id="nameDBField" type="text" name="nameDB" onChange={changeNameDB} value={nameDB} />
+          <label className={styles.label} htmlFor="nameConnectionField">CONNECTION NAME</label>
+          <Input id="nameConnectionField" type="text" name="nameConnection" onChange={changeNameConnection} value={nameConnection} />
+          <label className={styles.label} htmlFor="usernameField">USERNAME</label>
+          <Input id="usernameField" type="text" name="Username" onChange={changeUsername} value={username} />
+          <label className={styles.label} htmlFor="passwordField">PASSWORD</label>
+          <Input id="passwordField" type="password" name="password" onChange={changePassword} value={password} />
+          <label className={styles.label} htmlFor="typeField">TYPE</label>
+          <Select options={[{ name: 'PostgreSQL', value: '1' }]} classes={styles.select} value={type} onChange={changeType} />
```
Comment (VladaZhmurkevych) [inline src/components/newDBConnectionPopup/NewDBConnectionForm/index.js]: the same, move to the variable

**Review by VladaZhmurkevych — APPROVED** [review]

**Pull request merged**